packages/cluster/src/commands/up.js

```diff
@@ -5,13 +5,40 @@ const {
     initDockerComposeCache,
     makeComposeProject,
     makeDockerImage,
+    substituteVersion,
 } = require('../common')
+
+const defaults = require('../defaults')
 const { seed: doSeed } = require('../db')
 
-const run = async function({ v, port, seed, seedFile, update, ...argv }) {
+const run = async function({
+    name,
+    port,
+    seed,
+    seedFile,
+    update,
+    image,
+    dhis2Version,
+    root,
+    ...argv
+}) {
+    const { cluster } = argv
+
+    const resolvedVersion = dhis2Version || name
+    const resolvedImage = substituteVersion(
+        image || cluster.image || defaults.image,
+        resolvedVersion
+    )
+
+    const resolvedPort = port || cluster.port || defaults.port
+    const resolvedRoot = root || cluster.root || defaults.root
+    const resolvedContextPath = resolvedRoot ? '' : `/${name}`
```

Comment:
I like the simplicity of this but I worry about it.  It's a breaking change to default to a non-root context.  Should we default to the root context and have a `--no-root` flag (or something better) to switch to deploying on the probably less-common non-root context?

Reply:
Ok, I yield. The implementation docs uses the ROOT as well so let's go with that: https://docs.dhis2.org/2.32/en/implementer/html/install_server_setup.html#install_tomcat_dhis2_installation

Reply:
Rename it to `--custom-context`?